core-web/libs/portlets/edit-ema/portlet/src/lib/edit-ema-editor/edit-ema-editor.component.ts

```diff
@@ -1365,4 +1352,44 @@ export class EditEmaEditorComponent implements OnInit, OnDestroy {
                 this.placeItem(payload, dragItem);
             });
     }
+
+    /**
+     * Handle the inline editing event
+     *
+     * @param {*} { type, data }
+     * @return {*}
+     * @memberof EditEmaEditorComponent
+     */
+    #handleInlineEditingEvent({
+        type,
+        data
+    }: {
+        type: INLINE_EDITING_EVENT_KEY;
+        data?: InlineEditEventData;
+    }) {
+        if (!this.uveStore.isEnterprise()) {
+            this.#dotAlertConfirmService.alert({
+                header: this.dotMessageService.get('dot.common.license.enterprise.only.error'),
+                message: this.dotMessageService.get('editpage.not.lincese.error')
+            });
+
+            return;
+        }
+
+        switch (type) {
+            case 'BLOCK_EDITOR':
+                this.blockSidebar?.open(data);
+                break;
+
+            case 'WYSIWYG':
+                this.inlineEditingService.initEditor();
+                this.uveStore.setEditorState(EDITOR_STATE.INLINE_EDITING);
+                break;
+
+            default:
+                console.warn('Unknown block editor type', type);
+
+                break;
+        }
```

Comment:
Awesome

Reply:
There are some things that need review, such as `EDITOR_STATE.INLINE_EDITING` and the `inline-editing` service. I believe they are not needed. It's going to be handled in a separate ticket

Reply:
Also: I believe this and `CLIENT_ACTIONS` could be a service, this components it over 1000 lines, that's crazy

Reply:
Not sure if it should be a service, but maybe we can refactor this to reduce, lines in the component